pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/TableUpsertMetadataManager.java

```diff
@@ -43,4 +46,9 @@ public interface TableUpsertMetadataManager extends Closeable {
    * Stops the metadata manager. After invoking this method, no access to the metadata will be accepted.
    */
   void stop();
+
+  void waitTillReady()
```

Comment:
Add some javadoc for the new added method

Reply:
Do we want to make it more specific? e.g. `waitTillReadyToAddData` or `waitTillPreloadIsDone`

Reply:
delete, as no need for this method anymore as discussed below.